builder/optimizations.cc

```diff
@@ -55,6 +55,24 @@ bool is_copy(var src, expr src_x, int src_d, var dst, var dst_x, int dst_d, expr
     src_dim.fold_factor = dim::unfolded;
     return true;
   } else {
+    expr fold_factor = buffer_fold_factor(src, src_d);
+    bool is_mod = false;
+
+    // Try to parse src_x = dst_x % fold_factor
+    if (const class mod* m = src_x.as<class mod>()) {
+      const bool is_unfolded =
+          !src_dim.fold_factor.defined() ||
```

Comment:
I don't think all of the callers set this now, it just starts out undefined. Thinking about this, this is quite tricky. We might need to track fold factors in the visitor so we can know what they are.